prereise/gather/demanddata/transportation_electrification/generate_state_immediate_files.py

```diff
@@ -0,0 +1,53 @@
+import os
+
+from prereise.gather.const import abv2state
+from prereise.gather.demanddata.transportation_electrification import const
+from prereise.gather.demanddata.transportation_electrification.generate_BEV_vehicle_profiles import (
+    generate_bev_vehicle_profiles,
+)
+
+def write_state_demand_files(
+    demand_output, state, dir_path=None
+):
+    """Create files for each state
+
+    :param dict demand_output: 
+    :param str state: 
+    :param str dir_path: path to folder wher files will be written. Default to *'data/
+        regional_scaling_factors'* in current directory.
+    """
+    if dir_path is None:
+        dir_path = os.path.join(
+            os.path.dirname(os.path.abspath(__file__)),
+            "output",
+        )
+    os.makedirs(dir_path, exist_ok=True)
+    print(f"Writing state demand files in {dir_path}")
+
+    demand_output.to_pickle(
+        os.path.join(dir_path, f"immediate_{state}.pkl")
+    )
+
+
+if __name__ == "__main__":
+    for state in abv2state:
+        state_demand_df = generate_bev_vehicle_profiles(
+            vehicle_trip_data_filepath=os.path.join(
+                const.data_folder_path,
+                "nhts_census_updated_dwell.mat",
+            ),
+            charging_strategy="immediate",
+            veh_type="ldv",
+            veh_range=100,
+            projection_year=2024,
+            state=state,
+            power=6.6,
+            location_strategy=2,
+            trip_strategy=1,
+        )
+
+    print(state)
+
+    write_state_demand_files(state_demand_df, state)
+
+    print()
```

Comment:
Should the 3 lines above be moved within the `for` loop? Or you just want to print out and write data for last sate in `abv2state`? Also , can we do `print(f"{state}\n")` inthe first `print` and remove the last one?

Reply:
Yep, fixed this already, will push the changes